website/src/components/Visualisations/Resources/LineGraph.tsx

```diff
@@ -0,0 +1,143 @@
+import React from "react";
+import { Brush, ResponsiveContainer, LineChart, Line, YAxis, XAxis, CartesianGrid, Tooltip, Legend, Surface, Symbols, ReferenceLine } from 'recharts';
+import outputJSON from '../../../output/output.json';
+import _ from "lodash";
+
+
+class LineRechartComponent extends React.Component<{}, any> {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            disabled: [],
+            lineColours: {
+                team1: "#0095FF",
+                team2: "#FF0000",
+                team3: "#802FF0",
+                team4: "#00C49F",
+                team5: "#FFBB28",
+                team6: "#FF8042",
+            },
+            datapaths: {
+                team1: "ClientInfos.Team1.Resources",
+                team2: "ClientInfos.Team2.Resources",
+                team3: "ClientInfos.Team3.Resources",
+                team4: "ClientInfos.Team4.Resources",
+                team5: "ClientInfos.Team5.Resources",
+                team6: "ClientInfos.Team6.Resources",
+            },
+            chartData: outputJSON
+        };
+    }
+
+    handleClick = dataKey => {
+        const { disabled } = this.state;
+        if (disabled.includes(dataKey)) {
+            this.setState({
+                disabled: this.state.disabled.filter(obj => obj !== dataKey)
+            });
+        }
+        else {
+            this.setState({
+                disabled: this.state.disabled.concat(dataKey)
+            });
+        }
+    };
+
+    renderCustomizedLegend = ({ payload }) => {
+        return (
+            <div className="customized-legend">
+                {payload.map(entry => {
+                    const { dataKey, colour } = entry;
+                    const { disabled } = this.state;
+                    const active = disabled.includes(dataKey);
+                    const style = {
+                        marginRight: 10,
+                        colour: active ? "#AAA" : "#000"
+                    };
```

Comment:
Not important, but would like to discuss for fun:

I usually like declaring constants like this at the top, so in this case between l47 & l48. I feel it makes code a bit clearer that the constants aren't part of the imperative looping logic.